canisters/src/pair/principal_ext.rs

```diff
@@ -0,0 +1,18 @@
+use ic_cdk::export::candid::Principal;
+use async_trait::async_trait;
+
+#[async_trait]
+pub trait PairPrincipalExt {
+    /// Sets the liquidity cap for the pair. This method can only be called by the pair owner,
+    /// which is usually the pair factory.
+    async fn set_cap(&self, amount: Option<u128>) -> Result<(), String>;
```

Comment:
Just one question - is a get cap needed?

Reply:
Get cap is not used by factory, so no need for it in helpers. Pair canister has this method.